File: python/arrays/isValidAnagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(s):
            return False

        map ={}

        for char in s:
            map[char] = map.get(char,0) + 1

        for char in t:
            if char not in map:
                return False
            map[char] -= 1
            if map[char] == 0:
                del map[char]
        return  len(map) == 0

File: elixir/arrays/is_valid_anagram.exs
defmodule Solution do
  @spec is_anagram(s :: String.t(), t :: String.t()) :: boolean
  def is_anagram(s, t) when byte_size(s) != byte_size(t), do: false

  def is_anagram(s, t) do
    h =
      s
      |> String.graphemes()
      |> Enum.reduce(%{}, fn char, acc ->
        Map.update(acc, char, 1, &(&1 + 1))
      end)

    t
    |> String.graphemes()
    |> Enum.reduce_while({h, true}, fn char, {acc, _} ->
      case acc[char] do
        nil -> {:halt, {acc, false}}
        0 -> {:halt, {acc, false}}
        count -> {:cont, {Map.put(acc, char, count - 1), true}}
      end
    end)
    |> elem(1)
  end
end
